templates/minimal/runtime/Cargo.toml

```diff
@@ -21,6 +21,7 @@ polkadot-sdk = { workspace = true, features = [
 	"pallet-transaction-payment-rpc-runtime-api",
 	"runtime",
 ] }
+serde_json = { workspace = true, default-features = false }
```

Comment:
I am surprised that `alloc` feature is not required here. This was needed for other runtimes.

Reply:
I am seeing `templates/parachain/runtime` with the same recipe. Can you give me an example of a runtime that can not compile without the serde_json's `alloc` feature? There are a few reasons why not mentioning the `alloc` feature makes this still work.

[1] https://doc.rust-lang.org/rustc/platform-support/wasm64-unknown-unknown.html#requirements (wasm targets still have std working, with allocation support included based on dlmalloc)
[2] https://serde.rs/no-std.html#no-std-support (some other dependencies might depend on `serde_json` with features included for `std/alloc`, and compilation for the crate's `serde_json` dependency will be an union of all dependencies' features).
[3] Seems like most runtimes have a default feature called `std`, where `serde_json/std` is one of the dependency features.

Reply:
`alloc` for `serde_json` is enabled by other dependencies via `cargo` features resolution mechanism.

You can quickly check this by using:
`cargo tree -p minimal-template-runtime -i serde_json --no-default-features -e features`

or, try to do minimal app with `serde_json` w/o default features. I am getting following error.
```
error: serde_json requires that either `std` (default) or `alloc` feature is enabled
   --> /home/miszka/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde_json-1.0.128/src/lib.rs:368:1
    |
368 | / compile_error! {
369 | |     "serde_json requires that either `std` (default) or `alloc` feature is enabled"
370 | | }
```

I just recall that we had some problems when this feature was not specified (unfortunately I cannot recall what the problem was exactly, but it was tricky to find).

Reply:
Yes, so you suggest that is recommended to still add "alloc" anyways, to not depend in cargo features resolution that currently makes this working because of a dependency that uses serde_json with std/alloc. I can add it to all templates runtimes then.